### PR TITLE
feat(backend/shared): UAM (User Access Management) - Groups tm-230

### PR DESCRIPTION
**Pull request by YourTeaChan**

### CODE REVIEW

File: apps/backend/src/modules/groups/group.repository.ts [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+			.relate(permissionId)
+			.execute();
+
+		const permissions = await this.groupModel
+			.relatedQuery(RelationName.PERMISSIONS)
+			.for(groupId)
+			.castTo<PermissionModel[]>()
+			.execute();
+
+		return permissions.map((permission) =>
```
Comment (Vfivew) [inline apps/backend/src/modules/groups/group.repository.ts:38]: B27

File: packages/shared/src/modules/groups/libs/enums/group-error-message.enum.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const GroupErrorMessage = {
+	GROUP_ALREADY_EXISTS: "Group already exists.",
+	GROUP_ALREADY_HAS_THIS_PERMISSION: "Group already has this permission.",
+	GROUP_NOT_FOUND: "Group not found",
```
Comment (Vfivew) [inline packages/shared/src/modules/groups/libs/enums/group-error-message.enum.ts:4]: You misses dot
```suggestion
	GROUP_NOT_FOUND: "Group not found.",
```

**Review by Vfivew — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/groups/group.service.ts [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+class GroupService implements Service {
+	private groupRepository: GroupRepository;
+	private permissionService: PermissionService;
+	private userService: UserService;
+
+	public constructor(
+		groupRepository: GroupRepository,
+		permissionService: PermissionService,
+		userService: UserService,
```
Comment (tarmiel) [inline apps/backend/src/modules/groups/group.service.ts:26]: Pass it as an object and specify type Сonstructor

File: packages/shared/src/modules/groups/libs/enums/group-error-message.enum.ts (hunk 2 above)
Comment (tarmiel) [inline packages/shared/src/modules/groups/libs/enums/group-error-message.enum.ts:4]:
```suggestion
	GROUP_NOT_FOUND: "Group not found.",
```

File: apps/backend/src/modules/users/user.repository.ts [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import { type Repository } from "~/libs/types/types.js";
+import { GroupEntity } from "~/modules/groups/group.entity.js";
 import { UserEntity } from "~/modules/users/user.entity.js";
 import { type UserModel } from "~/modules/users/user.model.js";
 
+import { PermissionEntity } from "../permissions/permissions.js";
```
Comment (tarmiel) [inline apps/backend/src/modules/users/user.repository.ts:6]:
```suggestion
import { PermissionEntity } from "~/modules/permissions/permissions.js";
```

File: apps/backend/src/modules/users/user.entity.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import { type Entity } from "~/libs/types/types.js";
 
+import { type GroupEntity, type GroupResponseDto } from "../groups/groups.js";
```
Comment (tarmiel) [inline apps/backend/src/modules/users/user.entity.ts:3]:
```suggestion
import { type GroupEntity, type GroupResponseDto } from "~/modules/groups/groups.js";
```

**Review by tarmiel — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/permissions/permission.repository.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
+import { type Repository } from "~/libs/types/repository.type.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/permissions/permission.repository.ts:1]:
```suggestion
import { type Repository } from "~/libs/types/types.js";
```

File: apps/backend/src/modules/permissions/permission.service.ts [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+			throw new PermissionError({
+				message: PermissionErrorMessage.PERMISSION_NOT_FOUND,
+				status: HTTPCode.NOT_FOUND,
+			});
+		}
+
+		const { key, name } = permission;
+		const updatedPermission = await this.permissionRepository.update(
+			id,
+			PermissionEntity.initializeNew({ key, name }),
```
Comment (fshabanov) [inline apps/backend/src/modules/permissions/permission.service.ts:105]: What if with update we send key/name of an existing permission?

Comment (fshabanov) [reply apps/backend/src/modules/permissions/permission.service.ts:105]: Same for groups

File: apps/backend/src/modules/users/user.controller.ts [hunk 8]
```diff
@@ -55,6 +63,13 @@ class UserController extends BaseController {
 		});
 	}
 
+	private async findAll(): Promise<APIHandlerResponse> {
```
Comment (fshabanov) [inline apps/backend/src/modules/users/user.controller.ts:66]: Add swagger docs

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/permissions/permission.service.ts [hunk 9]
```diff
@@ -0,0 +1,112 @@
+import { HTTPCode } from "~/libs/enums/enums.js";
+import { type Service } from "~/libs/types/service.type.js";
```
Comment (Sytnikovakatya) [inline apps/backend/src/modules/permissions/permission.service.ts:2]:
```suggestion
import { type Service } from "~/libs/types/type.js";
```

**Review by GEOFARL — APPROVED** [review]

File: apps/backend/src/libs/hooks/check-user-permissions.hook.ts [hunk 10]
```diff
@@ -0,0 +1,61 @@
+import {
```
Comment (what1s1ove) [inline apps/backend/src/libs/hooks/check-user-permissions.hook.ts:1]: should be in a separate folder

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**